### PR TITLE
Change iterating through particles to iterating through cells

### PR DESCRIPTION
**Pull request by ryanwentaoxu**
remove cells by iterating through cells instead of iterating through particles

### CODE REVIEW

File: include/particle.tcc [hunk 1]
```diff
@@ -208,7 +208,7 @@ bool mpm::Particle<Tdim, Tnphases>::assign_cell_id(mpm::Index id) {
 template <unsigned Tdim, unsigned Tnphases>
 void mpm::Particle<Tdim, Tnphases>::remove_cell() {
   // if a cell is not nullptr
-  if (cell_ != nullptr) cell_->remove_particle_id(this->id_);
+  // if (cell_ != nullptr) cell_->remove_particle_id(this->id_);
```
Comment (kks32) [inline include/particle.tcc]: why is this commented out?

Comment (CassiniXu) [reply include/particle.tcc]: When iterate through cells, one needs to iterate through local paticles_ids in that cells and then delete them, but when calling this function once, the particles_ids would be deleted and the iteration process would be stopped. So I tried to comment it out.

Comment (CassiniXu) [reply include/particle.tcc]: We can write another similar function which comment this out and the original one can remain the same.

Comment (ryanwentaoxu) [reply include/particle.tcc]: > why is this commented out?

When iterate through cells, one needs to iterate through local paticles_ids in that cells and then delete them, but when calling this function once, the particles_ids would be deleted and the iteration process would be stopped. So I tried to comment it out.

Comment (kks32) [reply include/particle.tcc]: but this is just removing id

Comment (ryanwentaoxu) [reply include/particle.tcc]: > why is this commented out?

We can write another similar function which comment this out and the original one can remain the same.

**Pull request merged**